client/src/js/components/bhStockEntryExitType.js

```diff
@@ -0,0 +1,55 @@
+angular.module('bhima.components')
+    .component('bhStockEntryExitType', {
+        templateUrl: 'modules/templates/bhStockEntryExitType.tmpl.html',
+        controller: StockEntryExitTypeController,
+        bindings: {
+            onEntryExitTypeSelectCallback: '&',
+            reference: '<?',
+            displayName: '<?',
+            isEntry: '@'
+        },
+    });
+
+StockEntryExitTypeController.$inject = [];
+
+/**
+ * Stock Entry Exit Type component
+ */
+function StockEntryExitTypeController() {
+    var $ctrl = this;
+    $ctrl.selectedEntryExitType = null;
+    $ctrl.isEntry === 'true';
```

Comment:
Does this actually work?

I think you might be missing an assignment - this is just an evaluation.  Maybe...
```js
$ctrl.isEntry = $ctrl.isEntry === 'true'
```
?

Either way, this should probably go in an `$onInit` method to make sure the bindings are correctly initialized.